enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+25. The `openshift-install wait-for install-complete` process loses
+    the connection to the assisted service REST API and starts relying
+    entirely on the OpenShift API for data.
+26. The `openshift-install wait-for install-complete` process sees
+    that the `ClusterVersion` API in the cluster shows that the
+    deployment has completed, and reports success then exits.
+27. The orchestration tool may take steps to clean up (removing the
+    ISO, disconnecting it from the boot sequence for the VMs or BMCs,
+    etc.).
+
+TODO: Write the steps describing what happens when create-cluster sees
+      that a host has failed validation.
+
+### User Stories
+
+- As an OpenShift infrastructure owner, I need to be able to integrate
+  the installation of my first on-premises OpenShift cluster with my
+  automation flows and tools.
+- As an OpenShift infrastructure owner, I must be able to provide the
+  definition of the cluster I want to deploy as input to the tool.
+- As an OpenShift Infrastructure owner, I must be able to retrieve the
+  validation errors from an in-progress or failed deployment in a
+  programmatic way.
+- As an OpenShift Infrastructure owner, I must be able to get the
+  events and progress of the installation in a programmatic way.
+- As an OpenShift Infrastructure owner, I must be able to retrieve the
+  kubeconfig and OpenShift Console URL for my new cluster in a
+  programmatic way.
+
+### API Extensions
+
+This enhancement describes changes to the installer, but will not
+introduce any in-cluster API changes via CRDs, webhooks, or aggregated
+API servers.
+
+### Implementation Details/Notes/Constraints [optional]
+
+We already have several ways to describe OpenShift clusters for our
+different installation workflows, and want to avoid inventing a new
+one. Therefore we will reuse existing inputs relevant to on-premises
+deployment, borrowing from the IPI installer and other tools such as
+the zero-touch provisioning (ZTP) system in multi-cluster engine
+(MCE).
+
+Users often do not want to store the credentials for the baseboard
+management controllers (BMC) of their hosts inside of the cluster they
+are deploying. Therefore we cannot rely on them to give us those
+credentials during installation, and we must support tools outside of
+OpenShift managing the hosts' boot process. This places more of the
+burden of designing the end-to-end automated workflow on users or our
+solution architects.
+
+### Risks and Mitigations
+
+The inputs to building an on-premises cluster, especially the per-host
+static network settings, can be complex. Manually creating the inputs
+to the installer provides many opportunities to introduce errors. For
+the automated use case, we assume that users are likely to mitigate
+this risk by also automating production of the input files using
+templates or file generators. Other mitigation approaches, such as a
+GUI for constructing the files, are left to future enhancements.
```

Comment:
I think giving some thought to static validation of network configuration would be worthwhile, I know this is potentially difficult, but as a user if I provided some bad config (duplicate IPs, mismatched vlans, etc) I'd much rather learn about that early in the process vs at runtime when it's harder to debug.

/cc @cybertron

Reply:
If we're already doing something like that for the IPI installer, we would reuse it. If not, then I think we should do it independently of this enhancement.

Reply:
Zane's right. We are waiting to have the bindings by the end of the month.

Reply:
I added some details about static network validation. I think we're going to want a separate enhancement to describe how the use of tho Rust library will work.